index.js

```diff
@@ -131,14 +150,17 @@ class Ky {
 		this._options.prefixUrl = String(this._options.prefixUrl || '');
 		this._input = String(input || '');
 
-		if (this._options.prefixUrl && this._input.startsWith('/')) {
-			throw new Error('`input` must not begin with a slash when using `prefixUrl`');
-		}
-		if (this._options.prefixUrl && !this._options.prefixUrl.endsWith('/')) {
-			this._options.prefixUrl += '/';
+		const isFullUrlInput = isAbsoluteUrl(this._input);
+		const isFullUrlPrefix = isAbsoluteUrl(this._options.prefixUrl);
+		const documentBaseUri = _globalThis.document && _globalThis.document.baseURI;
+
+		if (!isFullUrlInput && !isFullUrlPrefix && !documentBaseUri) {
+			throw new Error('The `prefixUrl` option must be set to a full URL, incl. protocol, when using a relative `input` URL inside a non-browser environment');
 		}
 
-		const url = new _globalThis.URL(this._options.prefixUrl + this._input, document.baseURI);
+		const urlString = createRequestUrl(this._options.prefixUrl, this._input);
```

Comment:
Why cannot we just `new URL(this._input, this._options.prefixUrl)`?

Reply:
@szmarczak if we did, then `prefixUrl` would have to be an absolute URL, which is undesirable. I'm really quite annoyed that `new URL()` doesn't support relative URLs. I recommend avoiding `new URL()` when the desired behavior is to follow browser resolution rules. As of #74, we conditionally use it only if we need it.

Reply:
> if we did, then prefixUrl would have to be an absolute URL, which is undesirable.

Umm... But we do: https://github.com/sindresorhus/ky/blob/d5d3eaa24f903294d1b86d96a7652262c61536eb/index.js#L145-L151

>  I recommend avoiding new URL() when the desired behavior is to follow browser resolution rules. As of #74, we conditionally use it only if we need it.

I mean `_globalThis.URL` :P

Reply:
Is there any particular reason for `createRequestUrl`? Why we can't leave it as it is?